### PR TITLE
Fix malformed version number error for terraform and github actions

### PR DESCRIPTION
**Pull request by amazimbe**
### What are you trying to accomplish?
Stop github actions and terraform updates erroring for version formats like `v3.3.2` and `1.17.2+backport-1`

### How will you know you've accomplished your goal?
Malformed version number errors in terraform and github actions should stop

### Checklist

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: terraform/lib/dependabot/terraform/version.rb [hunk 1]
```diff
@@ -31,6 +34,15 @@ def self.new(version)
       def to_s
         @version_string
       end
+
+      sig { override.params(version: VersionParameter).returns(T::Boolean) }
+      def self.correct?(version)
+        version = version.gsub(/^v/, "") if version.is_a?(String)
```
Comment (abdulapopoola) [inline terraform/lib/dependabot/terraform/version.rb:40]: these 2 lines are repeated in `initialize`; is this an opportunity to extract them out?

File: terraform/spec/dependabot/terraform/version_spec.rb [hunk 2]
```diff
@@ -30,4 +30,26 @@
       it { is_expected.to eq "1.0.0-pre1" }
     end
   end
+
```
Comment (abdulapopoola) [inline terraform/spec/dependabot/terraform/version_spec.rb:33]: can we get test coverage for the initialize too?

**Comment by abdulapopoola** [comment]
Do you know how we handle invalid versions like guids? Do we throw an error that is handled E2E?

File: github_actions/lib/dependabot/github_actions/version.rb [hunk 3]
```diff
@@ -31,6 +31,8 @@ def self.remove_leading_v(version)
 
       sig { override.params(version: VersionParameter).returns(T::Boolean) }
       def self.correct?(version)
+        return false if version.to_s.empty?
```
Comment (abdulapopoola) [inline github_actions/lib/dependabot/github_actions/version.rb:34]: what happens if the string is `"         "`; I think `empty?` will be false in this case

File: terraform/lib/dependabot/terraform/version.rb (hunk 1 above)
Comment (amazimbe) [reply terraform/lib/dependabot/terraform/version.rb]: Moved to a method

File: github_actions/lib/dependabot/github_actions/version.rb (hunk 3 above)
Comment (amazimbe) [reply github_actions/lib/dependabot/github_actions/version.rb]: I'm now stripping the spaces

**Comment by amazimbe** [comment]
> Do you know how we handle invalid versions like guids? Do we throw an error that is handled E2E?

Yes, we throw an error and based on https://github.com/dependabot/dependabot-core/blob/main/updater/bin/update_files.rb#L48 it looks like that error is handled E2E

**Comment by abdulapopoola** [comment]
> > Do you know how we handle invalid versions like guids? Do we throw an error that is handled E2E?
> 
> Yes, we throw an error and based on https://github.com/dependabot/dependabot-core/blob/main/updater/bin/update_files.rb#L48 it looks like that error is handled E2E

I'm not sure a runfailure gets piped E2E to show in the interface I demonstrated in yesterday's sync. I think we'll need to introduce a brand new error that captures the version challenge and then ensure it is shown in Dependabot API.

@kbukum1 is about to do something similar and you can pair with him on this.

**Comment by abdulapopoola** [comment]
> > > Do you know how we handle invalid versions like guids? Do we throw an error that is handled E2E?
> > 
> > 
> > Yes, we throw an error and based on https://github.com/dependabot/dependabot-core/blob/main/updater/bin/update_files.rb#L48 it looks like that error is handled E2E
> 
> I'm not sure a runfailure gets piped E2E to show in the interface I demonstrated in yesterday's sync. I think we'll need to introduce a brand new error that captures the version challenge and then ensure it is shown in Dependabot API.
> 
> @kbukum1 is about to do something similar and you can pair with him on this.

I think the error handling can be handled in a follow-up PR (once @kbukum1 adds some examples). Approving to avoid holding up for now.

**Review by abdulapopoola — APPROVED** [review]

**Pull request merged**